tests/test_rcatype.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import numpy as np
+
+from bugbug.models.rcatype import RCATypeModel
+
+
+def test_get_rca_from_whiteboard():
+    model = RCATypeModel()
+    # Case 1: No rca
+    assert model.get_rca_from_whiteboard("[Whiteboard1][Not RCA type]") == []
+    # Case 2: RCA : A and RCA - A
+    assert model.get_rca_from_whiteboard("[RCA: cornercase]") == ["cornercase"]
+    assert model.get_rca_from_whiteboard("[rca - codingerror]") == ["codingerror"]
+    # Case 3: Multiple rca types
+    assert model.get_rca_from_whiteboard("[rca - cornercase][rca - codingerror]") == [
+        "cornercase",
+        "codingerror",
+    ]
+    assert model.get_rca_from_whiteboard("[rca : systemerror][rca - codingerror]") == [
+        "systemerror",
+        "codingerror",
+    ]
+    assert model.get_rca_from_whiteboard("[rca - cornercase][rca : testingerror]") == [
+        "cornercase",
+        "testingerror",
+    ]
+    assert model.get_rca_from_whiteboard("[rca : cornercase][rca : codingerror]") == [
+        "cornercase",
+        "codingerror",
+    ]
+    # Case 4: subcategories enabled, with rca already present in the list
+    model = RCATypeModel(rca_subcategories_enabled=True)
+    assert model.get_rca_from_whiteboard("[RCA: codingerror - syntaxerror]") == [
```

Comment:
You should also add a test where subcategories are disabled and the output for `[RCA: codingerror - syntaxerror]` is `codingerror`